### ISSUE TITLE
Screen Readers won't read punctuation characters in Shortcuts

### ISSUE THREAD

**Issue by franciscojma86** [flutter/flutter#142939]
### Steps to reproduce

1. Create a SingleActivator(LogicalKeyboardKey.comma, meta: true)
2. Create a MenuBarItem(shortcut: <the shortcut created above>
3. Using a Screen Reader (VoiceOver in my case), navigate to the menu item

### Expected results

The Screen Reader should read "Command Comma". Same with other punctuations like period, quotes, question mark, etc. 

### Actual results

The Screen Reader will read "Command [silent]", since the comma is silent.

### Code sample

<details open><summary>Code sample</summary>

```dart
final shortcut = SingleActivator(LogicalKeyboardKey.comma, meta: true);
MenuItemButton(
        shortcut: shortcut,
        onPressed: () {},
        child: Text('Test Widget)),
```

</details>

### Flutter Doctor output

[✓] Flutter (on macOS 14.3 23D56 darwin-arm64 (Rosetta), locale en)
    • Framework revision 457a95fac8 (3 days ago), 2024-02-02T00:00:00.000
    • Engine revision 9869d47a27
    • Dart version 5a5d4c2622

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b

[✓] Connected device (1 available)
    • macOS (desktop) • macos • darwin-arm64 • macOS 14.3 23D56 darwin-arm64 (Rosetta)

**Comment by huycozy** [comment]
@franciscojma86 How do you `navigate to the menu item`?

My keyboard doesn't have `meta` key so I use `control` instead but the reader doesn't read anything when I press `control` + `,`.

This is a complete sample code:

<details>
<summary>Sample code</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() async {
  runApp(
    const MaterialApp(
      title: 'Flutter Demo',
      home: MyApp(),
    ),
  );
}

class MyApp extends StatefulWidget {
  const MyApp({super.key});

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  final _shortcut = const SingleActivator(LogicalKeyboardKey.comma, meta: true);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: MenuItemButton(
          shortcut: _shortcut,
          onPressed: () {},
          child: const Text('Test Widget'),
        ),
      ),
    );
  }
}

```
</details>

 Please also share the entire output of `flutter doctor -v` as well. Thanks!

**Comment by gspencergoog** [comment]
> My keyboard doesn't have meta key so I use control instead but the reader doesn't read anything when I press control + ,.

The `meta` key is the Windows key on Windows, and the Command key (⌘) on macos.  There are some keyboards with an actual "Meta" key, but they produce the same keycode as the Windows key, so on Linux it's also the Windows key on most keyboards.

**Comment by huycozy** [comment]
Thank @gspencergoog, looks like I see the issue now. The reader will be stopped at `Command` without reading `,`. The same result on Windows with NVDA reader.

<details>
<summary>Demo</summary>

https://github.com/flutter/flutter/assets/104349824/a48cc470-f114-4f8f-8410-3131580d0d06
</details>

<details>
<summary>Sample code</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() async {
  runApp(
    const MaterialApp(
      title: 'Flutter Demo',
      home: MyApp(),
    ),
  );
}

class MyApp extends StatefulWidget {
  const MyApp({super.key});

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  final _shortcut = const SingleActivator(LogicalKeyboardKey.comma, meta: true);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: MenuItemButton(
          shortcut: _shortcut,
          onPressed: () {},
          child: const Text('Test Widget'),
        ),
      ),
    );
  }
}

```
</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.16.9, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.16.9 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 41456452f2 (14 hours ago), 2024-01-25 10:06:23 -0800
    • Engine revision f40e976bed
    • Dart version 3.2.6
    • DevTools version 2.28.5

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Applications/Android Studio Hedgehog.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.85.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.80.0

[✓] Connected device (4 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D                         • android-arm64  • Android 11 (API 30)
    • iPhone (mobile)  • d9a94afe2b649fef56ba0bfeb052f0f2a7dae95e • ios            • iOS 15.8 19H370
    • macOS (desktop)  • macos                                    • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome                                   • web-javascript • Google Chrome 120.0.6099.234

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.20.0-2.0.pre.17, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.20.0-2.0.pre.17 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6322fef7e0 (29 minutes ago), 2024-02-05 21:53:12 -0500
    • Engine revision 808886312e
    • Dart version 3.4.0 (build 3.4.0-104.0.dev)
    • DevTools version 2.32.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Applications/Android Studio Hedgehog.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.85.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.80.0

[✓] Connected device (4 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D                         • android-arm64  • Android 11 (API 30)
    • iPhone (mobile)  • d9a94afe2b649fef56ba0bfeb052f0f2a7dae95e • ios            • iOS 15.8 19H370
    • macOS (desktop)  • macos                                    • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome                                   • web-javascript • Google Chrome 121.0.6167.139

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>
